modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/plugins/rules-sidebar/components/RulesSidebar.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayForm from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import {
+	useConfig,
+	useForm,
+	useFormState,
+} from 'dynamic-data-mapping-form-renderer';
+import React, {useState} from 'react';
+
+import Sidebar from '../../../../js/components/sidebar/Sidebar.es';
+import RuleEditorModal from '../../../components/rules/RuleEditorModal';
+import RuleList from '../../../components/rules/RuleList';
+import {EVENT_TYPES} from '../../../eventTypes';
+
+function normalizeRule(rule) {
+	const _rule = {...rule};
+
+	_rule['logical-operator'] = _rule['logicalOperator'];
+	delete _rule.logicalOperator;
+
+	return _rule;
+}
+
+const RulesSidebar = ({title}) => {
+	const {ruleSettings} = useConfig();
+	const dispatch = useForm();
+	const {dataLayout} = useFormState({schema: ['dataLayout']});
+
+	const [rulesEditorState, setRulesEditorState] = useState({
+		isVisible: false,
+		rule: null,
+	});
+	const [keywords, setKeywords] = useState('');
+
+	const toggleRulesEditorVisibility = (rule) => {
+		setRulesEditorState((prevState) => ({
+			isVisible: !prevState.isVisible,
+			rule: rule && normalizeRule(rule),
```

Comment:
You can avoid saving the entire content of the rule in the state and just saving the `index` and when it is time to pass to the `RuleEditorModal` you can use the `index` to get it.

Reply:
Done